src/main/kotlin/ecommerce/services/CartItemServiceImpl.kt

```diff
@@ -88,10 +97,10 @@ class CartItemServiceImpl(
                 .findByProductIdAndMemberId(cartItemRequestDTO.productId, member.id!!)
                 ?: throw OperationFailedException("Cart item not found")
 
-        return cartItemRepository.save(existing.copy(quantity = cartItemRequestDTO.quantity))
-    }
+        if (cartItemRequestDTO.quantity <= 0) throw OperationFailedException("Quantity must be greater than zero")
+        if (existing.quantity == cartItemRequestDTO.quantity) return existing
 
-    override fun deleteAll() {
-        cartItemRepository.deleteAll()
+        existing.quantity = cartItemRequestDTO.quantity
```

Comment:
How about avoiding setters and getters in `CartItem` and creating methods inside it instead? That way, `CartItem` will have more responsibilities and stronger cohesion. 🙂